### ISSUE TITLE
Regarding Log2FC value

### ISSUE THREAD

**Issue by TK-Lab-Vanderbilt-University** [cailab-tamu/scTenifoldKnk#23]
Dear developers

Thank you for developing this wonderful tool. Thanks to you, we were able to obtain interesting results using the tool you developed.
However, I have a question regarding FC (Foldchange), the result obtained after Virtual KO analysis.

Is it theoretically okay to transform the FC value into log2FC by adding log2 ((ex) log2FC <- log2(FC))?
If it is okay to transform the value into log2FC, is it okay to classify it as an up-regulated gene or a down-regulated gene?

(ex)
log2(FC) > 1 = Up regulatory genes
log2(FC) < -1 = Down regulatory genes

Thank you

**Comment by Rohit-Satyam** [comment]
I don't think so that's would be valid. I raised a similar query [here](https://github.com/cailab-tamu/scTenifoldKnk/issues/13)  but the developers discourage inferring any directionality using LFC values. Besides, please read the paragraph from their paper below:

>Next, we note that scTenifoldKnk is designed to predict DR genes rather than DE genes. DR genes might be differentially expressed upon the gene KO. To examine the expression-level changes of virtual KO perturbed genes, we performed a systematic comparison between the scTenifoldKnk results and the results of DE analysis across the three [analyzed datasets](https://www.sciencedirect.com/topics/computer-science/analyzed-datasets): Trem2, Nkx2-1, and Hnf4ag. For each dataset, we started by computing the DE statistics for all genes. Specifically, we obtained the fold change (FC) of each gene’s expression in WT samples related to KO samples (WT/KO) using the DE analysis package MAST.[51](https://www.sciencedirect.com/science/article/pii/S2666389922000010#bib51) Then, we compared FC between significant DR genes and non-significant DR genes. We found that significant DR genes, or perturbed genes predicted by scTenifoldKnk, tend to have a larger FC value (p-value < 0.05 for all three cases [Trem2, Nkx2-1, and Hnf4ag], one-sided t-tests with log2-transformed FC values) than non-significant DR genes or non-perturbed genes ([Figure S4](https://www.sciencedirect.com/science/article/pii/S2666389922000010#mmc1)). **Thus, the expression of DR genes predicted by scTenifoldKnk is more likely to be downregulated in samples of the real KO experiments.**

I also tried their new tool called [GenKI](https://academic.oup.com/nar/article/51/13/6578/7184155) but is grossly slower than this tool and takes forever to run even on 112 CPUs. I really like the simplicity of this R package and how fast it is.

**Comment by dosorio** [comment]
Hi, thanks for using scTenifoldKnk, and sorry for the slow reply. @Rohit-Satyam is right. Thanks for jumping in.

The `FC` column in `diffRegulation` is not an expression fold change. For each gene, scTenifoldKnk measures the distance between its position in the WT network and its position in the KO network, after manifold alignment. `FC` is that distance squared, divided by the average squared distance across all genes:

```r
FC <- distance^2 / mean(distance^2)
```

It is the statistic behind the chi-square p-value, and it measures **how much** a gene's regulatory context changes after the KO, not **in which direction** its expression changes. So:

- `log2(FC)` can be computed, but it doesn't mean what log2FC means in a DE analysis.
- `log2(FC) > 1` means the gene is perturbed more than twice as much as the average gene.
- `log2(FC) < -1` means it is perturbed *less* than average, i.e. hardly affected. It does **not** mean down-regulated.
- Splitting genes into up- and down-regulated with this value isn't valid.

To call significant genes, use `p.adj` (e.g. FDR < 0.05), or rank genes by `Z` or `distance`.

scTenifoldKnk predicts **differentially regulated (DR)** genes: genes whose connections in the network change after the KO. It doesn't predict the direction of expression changes. As the paper notes, DR genes do tend to be down-regulated in real KO experiments, but that is a trend across genes, not a per-gene prediction.

If you want a rough idea of direction, look at the sign of the edges between the KO gene and each DR gene in the WT network (`output$tensorNetworks$WT`). Genes positively co-expressed with the KO gene are more likely to decrease after the KO, and negatively co-expressed genes are more likely to increase. Treat this as a hypothesis to test in real KO data or with an experiment, not as a result.

Hope this helps!
